Move cennznet/prml

cowboy-bebug opened the pull request:
Discarded commit histories as they are not relevant to plug.

Edit: the following wont' be moved:
- `prml/fees`: needs to be a separate refactor task
    - relates to https://github.com/cennznet/cennznet/issues/37)
    - has dependency on https://github.com/cennznet/plug-blockchain/blob/master/srml/support/src/fee.rs
- `prml/generic-asset`: will use the upstream merged `srml/generic-asset`

Closes #50

cowboy-bebug committed:
let there be light

jordy25519 reviewed (changes requested):
Requires changes following offline discussion to fix some tech debt

cowboy-bebug committed:
clean up attestation module
- add prml/attestation in root Cargo.toml workspace
- remove unused crates
- refactor crate aliasing for consistency

cowboy-bebug committed:
preserve original tab spacing

cowboy-bebug committed:
use local modules instead of repos for prml/attestation

jordy25519 commented on srml/support/test/tests/reserved_keyword/on_initialize.stderr:
```diff
@@ -1,47 +1,47 @@
   --> $DIR/on_initialize.rs:39:1
    |
-30 | reserved!(on_finalize on_initialize on_finalise on_initialise offchain_worker deposit_event);
+39 | reserved!(on_finalize on_initialize on_finalise on_initialise offchain_worker deposit_event);
```
Ah this is from doughnut changes, thought I had fixed this 😅

cowboy-bebug replied:
Yep I got ur back on this one 😃

cowboy-bebug committed:
change license header for attestation

cowboy-bebug committed:
remove fees and generic-asset prmls

cowboy-bebug committed:
update license header

jordy25519 commented on prml/attestation/src/lib.rs:
```diff
@@ -0,0 +1,139 @@
+//! Value is typically the hash of an off-chain attestation document but can be any value that fits the application's needs.
+//!
+//!
+//! We recommend that you use the Attestation JavaScript SDK when working with this module. The SDK can be found at https://github.com/cennznet/api.js/tree/master/packages/crml-attestation
```
I think we can remove this comment altogether

cowboy-bebug committed:
remove redundant lines

jordy25519 commented:
Approval granted the CI builds check out 😄
